js/main.js

```diff
@@ -191,9 +169,14 @@ function renderPokemonCard(pokemon) {
   $viewDiv.appendChild($heartIcon);
   $heartIcon.addEventListener('click', function addToFavorite(event) {
     var $heartIcon = document.querySelector('.heart-icon');
-    if (event.target.tagName === 'I') {
+    var liked = data.currentPokemon;
+    if (event.target.tagName === 'I' && event.target.classList.contains('favorite')) {
+      $heartIcon.classList.remove('favorite');
+      var favoriteEntryNums = data.favorites.map(pokemon => pokemon.entry_number);
+      var index = favoriteEntryNums.indexOf(liked.entry_number);
```

Comment:
You should change this to a simple `for` loop to find the matching object, like we discussed